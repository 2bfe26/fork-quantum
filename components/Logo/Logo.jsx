import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Image from '../Image';
import LogoCandidate from '../../static/logoB2c.svg';
import LogoTypes from './LogoTypes';

class Logo extends Component {
  static Business = LogoTypes.Business;

  static Candidate = LogoTypes.Candidate;

  static Education = LogoTypes.Education;

  static Google = LogoTypes.Google;

  static Facebook = LogoTypes.Facebook;

  render() {
    const { src, ...props } = this.props;
    return <Image src={src} {...props} />;
  }
}

Logo.defaultProps = {
  src: LogoCandidate,
  alt: 'Logo da Catho',
  width: 100,
  height: 50,
};

Logo.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
};

export default Logo;
