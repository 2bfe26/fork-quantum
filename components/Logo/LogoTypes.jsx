import React from 'react';
import Logo from './Logo';
import LogoCandidate from '../../static/logoB2c.svg';
import LogoBusiness from '../../static/logoB2b.svg';
import LogoEducation from '../../static/logoEducacao.svg';
import LogoGoogle from '../../static/logoGoogle.svg';
import LogoFacebook from '../../static/logoFacebook.svg';

const types = {
  Business: props => (<Logo src={LogoBusiness} alt="Catho Empresas" {...props} />),
  Candidate: props => (<Logo src={LogoCandidate} alt="Catho" {...props} />),
  Education: props => (<Logo src={LogoEducation} alt="Catho Educação" {...props} />),
  Facebook: props => (<Logo src={LogoFacebook} alt="Facebook" {...props} />),
  Google: props => (<Logo src={LogoGoogle} alt="Google" {...props} />),
};

types.Business.displayName = 'Logo.Business';
types.Candidate.displayName = 'Logo.Candidate';
types.Education.displayName = 'Logo.Education';
types.Facebook.displayName = 'Logo.Facebook';
types.Google.displayName = 'Logo.Google';

export default types;
