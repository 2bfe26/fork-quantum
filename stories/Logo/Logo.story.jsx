import React from 'react';
import PropTypes from 'prop-types';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import Logo from '../../components/Logo';
import LogoTypes from '../../components/Logo/LogoTypes';
import Heading from '../../.storybook/components/Heading';
import Atom from '../static/atom.svg';
import { TabbedView, Tab } from '../../.storybook/components/TabbedView';
import HowToImport from '../../.storybook/components/HowToImport';
import CodeExample from '../../.storybook/components/CodeExample';
import AutoPropsApi from '../../.storybook/components/AutoPropsApi';
import { Col, Row } from '../../components/Grid';
import GitlabLink from '../../.storybook/components/GitlabLink';

const StyledRow = styled(Row)`
  margin-bottom: 30px;
`;

const LogoExample = ({ name }) => {
  const Component = LogoTypes[name];
  return (
    <StyledRow key={`row-${name}`}>
      <Col phone={4}>
        <CodeExample
          code={`<Logo.${name} />`}
          showTitle={false}
        />
      </Col>
      <Col phone={4}>
        <Component />
      </Col>
    </StyledRow>
  );
};

storiesOf('7. Image', module)
  .add('Logo', () => {
    const logoList = Object.keys(LogoTypes);
    return (
      <Heading name="Logo" image={Atom}>
        <TabbedView>
          <Tab title="Usage">
            <HowToImport importModules="Logo" />
            <p>You can use these logos:</p>
            <ul>
              {logoList.map(name => <li key={name}>{name}</li>)}
            </ul>
            {logoList.map(name => (<LogoExample key={`example-${name}`} name={name} />))}
          </Tab>
          <Tab title="API">
            <AutoPropsApi component={Logo} ignoredProps={['src']} />
          </Tab>
        </TabbedView>
      </Heading>
    );
  });

LogoExample.propTypes = {
  name: PropTypes.string.isRequired,
};