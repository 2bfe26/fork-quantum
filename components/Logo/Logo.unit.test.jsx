import React from 'react';
import renderer from 'react-test-renderer';
import Logo from './Logo';

describe('Logo component ', () => {
  it('Should match snapshot with default logo', () => {
    const logo = <Logo alt="Example" />;
    expect(renderer.create(logo).toJSON()).toMatchSnapshot();
  });

  it('Should match snapshot with Candidate logo', () => {
    const logo = <Logo.Candidate />;
    expect(renderer.create(logo).toJSON()).toMatchSnapshot();
  });

  it('Should match snapshot with Business logo', () => {
    const logo = <Logo.Business />;
    expect(renderer.create(logo).toJSON()).toMatchSnapshot();
  });

  it('Should match snapshot with Education logo', () => {
    const logo = <Logo.Education />;
    expect(renderer.create(logo).toJSON()).toMatchSnapshot();
  });

  it('Should match snapshot with Facebook logo', () => {
    const logo = <Logo.Facebook />;
    expect(renderer.create(logo).toJSON()).toMatchSnapshot();
  });

  it('Should match snapshot with Google logo', () => {
    const logo = <Logo.Google />;
    expect(renderer.create(logo).toJSON()).toMatchSnapshot();
  });
});
