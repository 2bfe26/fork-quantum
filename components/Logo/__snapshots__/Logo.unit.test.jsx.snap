// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component  Should match snapshot with Business logo 1`] = `null`;

exports[`Logo component  Should match snapshot with Candidate logo 1`] = `null`;

exports[`Logo component  Should match snapshot with Education logo 1`] = `null`;

exports[`Logo component  Should match snapshot with Facebook logo 1`] = `null`;

exports[`Logo component  Should match snapshot with Google logo 1`] = `null`;

exports[`Logo component  Should match snapshot with default logo 1`] = `null`;
